packages/studio-web/src/app/file.service.ts

```diff
@@ -8,34 +8,24 @@ import { Injectable } from "@angular/core";
 export class FileService {
   constructor() {}
 
-  readFile$ = (blob: any) =>
-    Observable.create((obs: any) => {
-      if (!(blob instanceof Blob)) {
-        obs.error(new Error("`blob` must be an instance of File or Blob."));
-        return;
-      }
-
-      const reader = new FileReader();
-
+  readFile$(blob: Blob | File): Observable<string> {
+    const reader = new FileReader();
+    return Observable.create((obs: any) => {
       reader.onerror = (err) => obs.error(err);
       reader.onabort = (err) => obs.error(err);
       reader.onload = () => obs.next(reader.result);
       reader.onloadend = () => obs.complete();
-      return reader.readAsText(blob);
+      reader.readAsText(blob);
     });
-  readFileAsData$ = (blob: any) =>
-    Observable.create((obs: any) => {
-      if (!(blob instanceof Blob)) {
-        obs.error(new Error("`blob` must be an instance of File or Blob."));
-        return;
-      }
-
-      const reader = new FileReader();
-
+  }
```

Comment:
Nice - these are cleaner now.

Reply:
thanks, I found the gist where you got them from originally and noticed that it was actually somewhat wrong :)